### DISCUSSION TITLE
Добавить фильтр по протоколy javascript: ( bookmarklets ) chrome: , возможно список исключений

### DISCUSSION THREAD

**Discussion by kobav14** [k-ivan/visual-bookmarks-chrome#196] [Ideas]
Спасибо за отличный add-on!

Идея вкратце - добавить фильтр в настройки какие закладки или протоколы нужно пропускать.

Закладки пользователя могут содержать bookmarklets c протоколами javascript: chrome: etc. с которыми приложение работает некорректно как при отображении/редактировании, так и при генерации thumbnails (виснет при попытке загрузить миниатюру такой закладки, никогда не доходя до окончания обновления всех миниатюр). 
Так-же хорошо-бы иметь в настройках пользовательский black list, чтобы приложение не обрабатывало какие-то конкретные закладки.

![vb00](https://github.com/user-attachments/assets/219d8eeb-7c3d-492e-84d3-6c1c05d33425)
![vb01](https://github.com/user-attachments/assets/ed2afa62-b82f-4eb9-b783-f7a1af262e1d)

**Comment by k-ivan** [comment]
Здравствуйте!
Спасибо за идею.

Я не думал, что браузер позволяет создавать закладки с js логикой вместо урл )

Я проверю поведение генерации скриншотов и их зависания на протоколах chrome. Ссылки chrome должны работать, вроде как добавлял их обработку.

А вот js выполняться не будет, браузер блокирует подобные вещи внутри расширения. Скриншот само собой тоже не получится сделать.

**Comment by kobav14** [comment]
закладки с протоколом chrome: не вешают обновление в отличие от
javascript:, но идея в том что не всегда нужно видеть отображение всех
закладок, ведь могут быть закладки которые не открываются без VPN, или
что-то ещё, что пользователь захочет поместить в блэк лист

**Reply by k-ivan** [reply]
Попробую исключить JavaScript урл из генерации скринов. 

По поводу фильтров, по скрытию подумаю, мне кажется, что сам фильтр действительно  может быть полезен, но не по протоколам, а возможно по самим урл, здесь надо будет подумать.
